fluent/migrate/validator.py

```diff
@@ -19,12 +19,14 @@ class BadContextAPIException(Exception):
 
 
 def process_assign(node, context):
-    if isinstance(node.value, ast.Str):
-        val = node.value.s
+    if isinstance(node.value, ast.Constant):
```

Comment:
I'm lost here, and I feel like I'm missing something obvious. If I try localle, I see that `ast` has `ast.Constant`, but where the heck is that coming from?

```
>>> import ast
>>> print(ast.Constant)
<class 'ast.Constant'>
>>> vars(ast.Constant)
mappingproxy({'_fields': ('value', 'kind'), '__match_args__': ('value', 'kind'), '__module__': 'ast', '__doc__': 'Constant(constant value, string? kind)', 'kind': None, 'n': <property object at 0x104fc9cb0>, 's': <property object at 0x104fca070>})
```

Also, why this change, given Fluent hasn't changed in a while?

Reply:
This change is due to `ast.Str` being [deprecated](https://bugs.python.org/issue26146) in all the versions of Python we now support, and is due to be removed in a near-future release:
> Changed in version 3.8: Class [ast.Constant](https://docs.python.org/3/library/ast.html#ast.Constant) is now used for all constants.

The `isinstance()` usage works because `ast.Str` [inherits](https://github.com/python/cpython/blob/5d7b90db0cd463e69d15acf5498232b9bc2bfeaa/Lib/ast.py#L594) from `ast.Constant`.

> Also, why this change, given Fluent hasn't changed in a while?

Because this is about parsing Python, and it is constantly changing.

Reply:
Thanks, I missed this was Python's ast and not Fluent's.